### ISSUE TITLE
Dependency conflicts for the firebase codelab

### ISSUE THREAD

**Issue by samindaw** [flutter/flutter#128652]
Hi,
I;ve being following your codelab at https://firebase.google.com/codelabs/firebase-get-to-know-flutter and ran in to some issues when trying to add the dependencies mentioned in 4 [Configure Firebase](https://firebase.google.com/codelabs/firebase-get-to-know-flutter#3). Seems the latest versions of the packages end up conflicting. I ended up using the dependency versions mentioned in Step 04 [github](https://github.com/flutter/codelabs/blob/main/firebase-get-to-know-flutter/step_04/pubspec.yaml)

**Comment by dam-ease** [comment]
Thanks for filing this issue @samindaw. I am unable to reproduce this issue. I was able to add the latest versions of the packages following the instructions in the codelab. Are you running on the latest stable version of flutter?
Also, what errors do you seem to be getting??

<details><summary>pubspec.yaml file</summary>
<p>

```
  firebase_core: ^2.13.1
  firebase_auth: ^4.6.2
  cloud_firestore: ^4.8.0
  provider: ^6.0.5
  firebase_ui_auth: ^1.4.2
``` 

</p>
</details> 

<details><summary>flutter doctor -v</summary>
<p>

```
[✓] Flutter (Channel stable, 3.10.4, on macOS 13.4 22F66 darwin-arm64, locale
    en-NG)
    • Flutter version 3.10.4 on channel stable at /Users/dammya/sdks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 682aa387cf (6 days ago), 2023-06-05 18:04:56 -0500
    • Engine revision 2a3401c9bb
    • Dart version 3.0.3
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version
    32.1.0-rc1)
    • Android SDK at /Users/dammya/Library/Android/sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • ANDROID_HOME = /Users/dammya/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2022.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.78.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.50.0

[✓] VS Code (version 1.78.0)
    • VS Code at /Users/dammya/.Trash/Visual Studio Code 20.07.05.app/Contents
    • Flutter extension version 3.50.0

[✓] Connected device (4 available)
    • Damilola’s iPhone (mobile) • 71c843fa5f3dec767999f36224653c0faa78660e •
      ios            • iOS 16.5 20F66
    • iPhone 14 (mobile)         • 5FE8DE93-0F4C-45BA-8BED-BC9CD69B35D1     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-arm64   • macOS 13.4 22F66 darwin-arm64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 114.0.5735.106

[✓] Network resources
    • All expected network resources are available.

• No issues found!
``` 

</p>
</details>

**Comment by samindaw** [comment]
Thanks @dam-ease for checking this out. 

```
$ flutter --version
Flutter 3.10.1 • channel stable • https://github.com/flutter/flutter.git
Framework • revision d3d8effc68 (4 weeks ago) • 2023-05-16 17:59:05 -0700
Engine • revision b4fb11214d
Tools • Dart 3.0.1 • DevTools 2.23.1
```

To be more precise, I couldn't add the last dependency package `firebase_ui_auth` mentioned in the codelab. 

pubspec.yaml file before trying to install `firebase_ui_auth`.

```
  cupertino_icons: ^1.0.2
  google_fonts: ^5.0.0
  go_router: ^8.0.1
  firebase_core: ^2.13.1
  firebase_auth: ^4.6.2
  cloud_firestore: ^4.8.0
  provider: ^6.0.5
```
The error I get when trying to add the dependency `firebase_ui_auth` 
```
$ flutter pub add firebase_ui_auth
...
...
Because google_fonts >=4.0.5 depends on http ^1.0.0 and every version of desktop_webview_auth depends on http ^0.13.3, google_fonts >=4.0.5 is incompatible with desktop_webview_auth.
Thus, google_fonts >=4.0.5 is incompatible with firebase_ui_auth.
So, because gtk_flutter depends on both google_fonts ^5.0.0 and firebase_ui_auth any, version solving failed.
```

I downgraded `google_fonts` to a 2.x.x version which allowed me to install all the dependency. However when trying to compile and deploy on my android emulator I ran in to runtime errors. That's when I just used the ones mentioned in `step 4` repo which didn't give me any issues in resolving the dependencies, or any runtime errors.

ps: I'm on linux mint 20.3

**Comment by dam-ease** [comment]
Thanks for the detailed explanation @samindaw. This is not a bug or an issue with the codelab. Not all versions of all packages are compatible with each other, you just need to use a compatible version in your pubspec as I see you ended up doing.

**Comment by samindaw** [comment]
yes correct @dam-ease. Apologies if i had reported the problem I faced incorrectly. 

If I may make a suggestion, have the instructions updated to something like
```
$ flutter pub add cupertino_icons google_fonts go_router firebase_core firebase_auth cloud_firestore provider firebase_ui_auth
```
_(note I have also included the packages that were hardcoded in the original `pubspec.yaml` file found in step2 github)_
so that flutter would resolve the best combination of compatible versions for all dependencies all at once rather than trying resolve one at a time based on already resolved packages.
